File: Core/Entities/Payment/Transaction.cs
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.Dynamic;

namespace Core.Entities.Payment
{
    public class Transaction
    {
        public string basketId { get; set; }
        public string acceptance_token { get; set; }
        public int? amount_in_cents { get; set; }
        public string currency { get; set; }
        public string customer_email { get; set; }
        [Required]
        public PaymentMethod payment_method { get; set; }
        public string reference
        { get; set; }
        public CustomerData customer_data { get; set; }
        public ShippingAddress shipping_address { get; set; }
    }

    public class PaymentMethod
    {
        public string type { get; set; }
        public string payment_description { get; set; }
        public string user_type { get; set; }
        public string sandbox_status { get; set; }
        public int? installments { get; set; }
        public string token { get; set; }
        public string phone_number { get; set; } = "0";
        public string user_legal_id_type { get; set; }
        public string user_legal_id { get; set; } = "0";
        public string financial_institution_code { get; set; } = "0";
    }

    public class CustomerData
    {
        public string phone_number { get; set; }
        public string full_name { get; set; }
    }

    public class ShippingAddress
    {
        public string address_line_1 { get; set; }
        public string country { get; set; }
        public string region { get; set; }
        public string city { get; set; }
        public string name { get; set; }
        public string phone_number { get; set; }
    }
}

File: client/src/app/shared/models/transaction.ts
import { environment } from "src/environments/environment";
import { IAddress } from "./address";

export interface ITransaction {
    id?: string;
    created_at?: string;
    status?: string;
    payment_method_type?: string;
    payment_link_id?: any;
    basketId?: string;
    shipToAddress?: IAddress;
    
    acceptance_token: string;
    amount_in_cents: number;
    currency: string;
    customer_email: string;
    reference: string;
    payment_method: any;
    redirect_url: string;
    customer_data?: ICustomerData;
    shipping_address?: IShippingAddress;
}
  
export interface IShippingAddress {
    address_line_1?: string;
    city?: string;
    name?: string;
    phone_number?: string;
}
  
export interface ICustomerData {
    phone_number: string;
    full_name: string;
}
  

export class Transaction implements ITransaction{
    acceptance_token: string = "";
    amount_in_cents: number = 0;
    currency: string = "";
    customer_email: string = "";
    reference: string = "";
    redirect_url: string = `${environment.clientUrl}checkout/pedido`;

    basketId: string = "";
    payment_method: any = {};
    shipping_address: IShippingAddress;
    customer_data: ICustomerData;

    constructor(){
        this.shipping_address = new ShippingAddress();
        this.customer_data = new CustomerData();
    }
}

export class CustomerData implements ICustomerData{
    phone_number: string = "";
    full_name: string = "";
}

export class ShippingAddress implements IShippingAddress{
    address_line_1: string = "";
    city: string = "";
    name: string = "";
    phone_number: string = "";
}

export interface ICreditCardData {
    number: string;
    cvc: string;
    exp_month: string;
    exp_year: string;
    card_holder: string;
}